### DISCUSSION TITLE
Is there an IM command to tell for sure if IM was used to encipher a file?

### DISCUSSION THREAD

**Discussion by ZjYwMj** [ImageMagick/ImageMagick#6043] [Help] [answered]
### ImageMagick version

Most newest IM version

### Operating system, version and so on

Any

### Description

Is there something like
```
magick -IM_was_used_to_encipher any_file
```
which will return true or false, with a 100% correct answer?
It is true that there can be an enciphered image embedded within a top level visible image. I am asking about the top level image.
Alternatively, due to sometimes it is undesirable to reveal that a file is an enciphered image, perhaps there should be a switch that will enable the behavior in question. That is
```
magick visible_image -encipher password -when_asked_tell_IM_enciphered enciphered_image
```
?

**Comment by snibgo** [comment, marked as answer]
> Is there something like
magick -IM_was_used_to_encipher any_file
which will return true or false, with a 100% correct answer?

I don't see how that could be possible.

We can use:
```
magick toes.png -encipher password.txt x.png

magick x.png -verbose info:
```
The verbose info from x.png contains:
```
    cipher:mode: CTR
    cipher:nonce: 3e885eb7997736ceb4e6d0318b42bc42871860bfb623d5476662eeea9ae2ab41
    cipher:type: AES
```
However, I expect that could be removed by exiftool or similar. Likewise it could be spoofed.

**Reply by ZjYwMj** [reply]
Thank you.
I do see now there can not be an always correct answer.
Still, I didn't think earlier about ``-verbose info``. So you did answer my question. I mean, the unix [file](https://linux.die.net/man/1/file) command is regraded as a useful tool, despite its answers can probably also gets spoofed in a similar manner.

**Comment by urban-warrior** [comment]
You can determine if an image was encrypted by checking the distribution of the pixel components.  If it exceeds a certain threshold, the data could be considered random and a possible candidate for encryption.    You can still get false positives such as images created with plasma:fractal.  And you don't know for sure if ImageMagick encrypted the image or some other utility.   A fairly confident way to know is to look for properties that are in the `cipher` namespace.  However, not all image formats supports the `cipher:` tag.
